src/include/txn_inline.h

```diff
@@ -392,38 +392,97 @@ __wt_txn_op_delete_apply_prepare_state(WT_SESSION_IMPL *session, WT_REF *ref, bo
     WT_REF_UNLOCK(ref, previous_state);
 }
 
+/*
+ * __txn_op_delete_commit_verify_page_del_timestamp --
+ *     Verify the timestamps before setting it to the page delete structure.
+ */
+static WT_INLINE int
+__txn_op_delete_commit_verify_page_del_timestamp(
+  WT_SESSION_IMPL *session, WT_TXN_OP *op, bool locked)
+{
+    WT_ADDR_COPY addr;
+    WT_DECL_RET;
+    WT_PAGE_DELETED *page_del;
+    WT_REF *ref;
+    WT_REF_STATE previous_state;
+    WT_TXN *txn;
+    bool addr_found;
+
+    ref = op->u.ref;
+    txn = session->txn;
+    page_del = ref->page_del;
+
+    previous_state = WT_REF_GET_STATE(ref);
+    if (!locked)
+        WT_REF_LOCK(session, ref, &previous_state);
+
+    if (page_del != NULL) {
+        /* Validate the commit timestamp against the maximum durable timestamp on the page. */
+        WT_ENTER_GENERATION(session, WT_GEN_SPLIT);
```

Comment:
Are we holding a page index here? Why this is needed? If it is needed, should the caller do it instead of here?

Reply:
We need to hold the `WT_GEN_SPLIT` generation to avoid freeing the `ref->addr` underneath us while we are accessing it.

Reply:
What if it is already freed before we enter the generation?

Reply:
The `ref->addr` cannot be freed as it have uncommitted updates. But we have asserts to confirm that the access to the `ref->addr` must be within the `GEN_SPLIT`.

Reply:
Can you add a comment here?